doc/exts/pylint_messages.py

```diff
@@ -0,0 +1,227 @@
+            "refactor",
+            "information",
+        ):
+            messages = sorted(messages_dict[category], key=lambda item: item.name)
+            old_messages = sorted(old_messages_dict[category], key=lambda item: item[0])
+            messages_string = "".join(
+                f"   {category}/{message.name}.rst\n" for message in messages
+            )
+            old_messages_string = "".join(
+                f"   {category}/{old_message[0]}.rst\n" for old_message in old_messages
+            )
+
+            # Write list per category
+            stream.write(
+                f"""{get_rst_title(category.capitalize(), "-")}
+All messages in the {category} category:
+
+.. toctree::
+   :maxdepth: 2
+   :titlesonly:
+
+{messages_string}
+All renamed messages in the {category} category:
+
+.. toctree::
+   :maxdepth: 1
+   :titlesonly:
+
+{old_messages_string}
+
+"""
+            )
+
+
+def _write_redirect_pages(old_messages: OldMessagesDict) -> None:
+    """Create redirect pages for old-messages."""
+    for category, old_names in old_messages.items():
+        category_dir = os.path.join(PYLINT_BASE_PATH, "doc", "messages", category)
+        if not os.path.exists(category_dir):
+            os.makedirs(category_dir)
+        for old_name, new_names in old_names.items():
+            old_name_file = os.path.join(category_dir, f"{old_name[0]}.rst")
+            with open(old_name_file, "w", encoding="utf-8") as stream:
+                new_names_string = "".join(
+                    f"   ../{new_name[1]}/{new_name[0]}.rst\n" for new_name in new_names
+                )
+                stream.write(
+                    f""".. _{old_name[0]}:
+
+{get_rst_title(" / ".join(old_name), "=")}
+"{old_name[0]} has been renamed. The new message can be found at:
+
+.. toctree::
+   :maxdepth: 2
+   :titlesonly:
+
+{new_names_string}
+"""
+                )
+
+
+# pylint: disable-next=unused-argument
+def build_messages_pages(app: Optional[Sphinx]) -> None:
+    """Overwrite messages files by printing the documentation to a stream.
+    Documentation is written in ReST format.
+    """
+    # Create linter, register all checkers and extensions and get all messages
+    linter = PyLinter()
+    _register_all_checkers_and_extensions(linter)
+    messages, old_messages = _get_all_messages(linter)
+
+    # Write message and category pages
+    _write_message_page(messages)
+    _write_messages_list_page(messages, old_messages)
+
+    # Write redirect pages
+    _write_redirect_pages(old_messages)
+
+
+def setup(app: Sphinx) -> None:
+    """Connects the extension to the Sphinx process"""
+    # Register callback at the builder-inited Sphinx event
+    # See https://www.sphinx-doc.org/en/master/extdev/appapi.html
+    app.connect("builder-inited", build_messages_pages)
+
+
+if __name__ == "__main__":
+    pass
+    # Uncomment to allow running this script by your local python interpreter
+    # build_messages_pages(None)
```

Comment:
Why do we need to comment it ?

Reply:
Isn't it a good thing as it makes sure that the generated doc is up to date ?

Reply:
That is take care of by `doc/Makefile` 😄